docs/zkapps/faq.mdx

```diff
@@ -27,18 +26,16 @@ for (let y of [5, 10, 15]) {
 }
 ```
 
-You can think of using higher-level JS constructs as a form of metaprogramming, to stitch together the right SnarkyJS commands more easily.
-
-However, beware that you can't use _the value of Field elements_ to determine what constraints to add. Example: The following statement can't be used, if `b` is a `Bool`:
+It's fine to use JS language constructs such as the for-loop above to stitch together the right SnarkyJS commands. But you can't use the value of Field elements to determine what constraints to add. The following statement won't work, even though `b` is a SnarkyJS `Bool`:
 
 ```ts
 // bad
 if (b.toBoolean()) x.assertEquals(10);
 ```
 
-This doesn't work for two reasons:
+This example fails for two reasons:
 
-1. Fundamentally, your methods must be written such that they create the same constraints every time. A proof cannot be verified against the verification key for a different set of constraints. The code above, however, tried to add an `assertEquals` contraint depending on the value of `b`.
-2. More practically, you can't read out the values of variables in circuit code - an expression like `b.toBoolean()` will throw an error during `SmartContract.compile()` (which generates the prover and verification keys). This is because during `compile()`, variables like `b` don't have any JS values attached to them; they represent abstract variables used to build up an abstract arithmetic circuit.
+1. Circuit code has to interact with SnarkyJS built-in types. An expression like `b.toBoolean()` will throw an error during `SmartContract.compile()` (which generates the prover and verification keys) because you can't read out the values of variables like that. This is because during `compile()`, variables like `b` don't have any JS values attached to them; they represent abstract variables used to build up an abstract arithmetic circuit.
```

Comment:
I'm not sure the first sentence "Circuit code has to interact with SnarkyJS built-in types." does a good job of summarizing this bullet point.

I'd consider the following:

> 1. `b.toBoolean()` can't be used in circuit code at all. It will throw an error during `SmartContract.compile()` because ...

Reply:
Cool, yes I agree and I like your edit